api/src/Mail/Account.php

```diff
@@ -467,7 +467,10 @@ public function smtpTransport()
 			// Horde use locale for translation of error messages
 			Api\Preferences::setlocale(LC_MESSAGES);
 
-			if (!empty($GLOBALS['egw_info']['server']['sendmail']))
+			$api_config = Api\Config::read('phpgwapi');
+
+			if (!empty($GLOBALS['egw_info']['server']['sendmail']) ||
+				!empty($api_config['sendmail']))
 			{
```

Comment:
It's the same thing, first thing EGroupware does is to read what you call DB configuration into $GLOBALS['egw_info']['server'] to the stuff already there from the header.inc.php.

So please remove/revert that, as it is unnecessary.